### PR TITLE
Allow for multiple deployment YAML paths in apply CLI

### PR DESCRIPTION
**Pull request by cicdw**
### Summary

Allows for multiple paths to be specified and removes the non-working default (**cc:** @zangell44 )

### CODE REVIEW

File: src/prefect/cli/deployment.py [hunk 1]
```diff
@@ -258,60 +258,58 @@ def _load_deployments(path: Path, quietly=False) -> PrefectObjectRegistry:
...
+            deployment = DeploymentYAML(**data)
+            app.console.print(f"Successfully loaded {deployment.name!r}", style="green")
+        except Exception as exc:
+            exit_with_error(f"'{path!s}' did not conform to deployment spec: {exc!r}")
+
+        async with get_client() as client:
+            # prep IDs
+            flow_id = await client.create_flow_from_name(deployment.flow_name)
+
+            deployment.infrastructure = deployment.infrastructure.copy()
```
Comment (zangell44) [inline src/prefect/cli/deployment.py:286]: can someone link deployment infrastructure to a stable block?
im confused here
(no change in logic im just curious)

**Review by zangell44 — APPROVED** [review]

File: src/prefect/cli/deployment.py (hunk 1 above)
Comment (madkinsz) [reply src/prefect/cli/deployment.py:286]: I had planned for that to be the case eventually, but needed to set the run command and image tag (for container based runs) at deployment creation time which required a copy. If we have the manifest attached to the deployment now separately, we could probably load those fields from there at runtime instead and just override whatever is on the block to support a stable block. It gets a little confusing though.

**Comment by madkinsz** [comment]
Note if creation of one fails it'll just stop at that failure. I think we should separate the load and apply steps to minimize the chance of this occurring e.g.

instead of

- Load deployment
- Register deployment
... repeat

do

- Load all deployments given, which performs validation on the input YAML definitions
- Register all deployments given

**Comment by cicdw** [comment]
@madkinsz would you like me to address that in this PR or a follow up?

**Comment by madkinsz** [comment]
It sounds like this fixes a broken experience so I'm fine with a follow-up!

**Review by anna-geller — APPROVED** [review]
Works!

image

**Review by bunchesofdonald — APPROVED** [review]

**Pull request merged**